lua/rust-tools/inlay_hints.lua

```diff
@@ -4,7 +4,7 @@ local M = {}
 
 function M.new()
   M.namespace = vim.api.nvim_create_namespace("textDocument/inlayHints")
-  local self = setmetatable({ cache = {}, enabled = false }, { __index = M })
+  local self = setmetatable({ cache = {} }, { __index = M })
```

Comment:
```suggestion
  local self = setmetatable({ cache = {}, enabled = false }, { __index = M })
```

Reply:
Does it work because any lookup to the table entry will be falsey by default, therefore we don't need to do defaulting to false explicitly? This is my first Lua PR which is why this looks interesting. These suggestions seem to implicate that we can add these values dynamically in the self table and it will work just fine?

Reply:
Yep, it will work. You can check this commit to try it out: https://github.com/MunifTanjim/rust-tools.nvim/commit/2ba50c9baee2b9ca4456e8c3dd59580182337d5c

But yeah... it's a bit confusing to be honest. There's room for improvements (re-organizing some stuffs to make it easier to understand).

Reply:
I see. This makes sense as most minimal working solution, so I am gonna commit that.